### ISSUE TITLE
Improve search mechanism, allow mistakes in search word 

### ISSUE THREAD

**Issue by matthieu-rolland** [PrestaShop/PrestaShop#14588]
**Is your feature request related to a problem? Please describe.**

This feature request is related to this issue:
https://github.com/PrestaShop/PrestaShop/issues/12407

**There are two problems with the product search mechanism:**

Search words are stored in a table called 'search_word':

1/ Our word length in the search_word table is fixed, so looking for a product whose name is longer than its 'word' column's length won't work if we type the word entirely or even only the end of the word, since the word is truncated in the search_word table.

2/ If we make a mistake in the search term, ex: appre instead of apple, then the search won't work.
 
**Describe the solution you'd like**
A solution that could mitigate both those issues would be to implement a [Levenshtein distance](https://www.google.com/search?ei=KAYnXY6QMcSLlwS3mrqoAQ&q=levenshtein+distance&oq=levenhstein&gs_l=psy-ab.3.0.0i10l10.51908.52944..55068...0.0..0.75.645.11......0....1..gws-wiz.......0i71j0i67j0.nLv1gbrdOAc) mechanism.

Basically we could create a Levenhstein search function in mysql, this function would be created during the prestashop install process (after creating the database).

Here is an [example](https://www.lucidar.me/en/web-dev/levenshtein-distance-in-mysql/) of how it is implemented in mysql.

Also checkout this [stackoverflow](https://stackoverflow.com/questions/4671378/levenshtein-mysql-php) .

The use of this mechanism could be made optional, by making it configurable in the backoffice.

**Describe alternatives you've considered**
An easier alternative would be to set a much bigger column length to search_word's word column.

**Comment by khouloudbelguith** [comment]
Ping @PrestaShop/prestashop-core-developers what do you think?

Thanks!
